sample-resource-plugin/src/main/java/org/opensearch/sample/resource/actions/transport/GetResourceTransportAction.java

```diff
@@ -0,0 +1,173 @@
+
+import org.opensearch.OpenSearchStatusException;
+import org.opensearch.ResourceNotFoundException;
+import org.opensearch.action.get.GetRequest;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.common.Nullable;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.common.xcontent.LoggingDeprecationHandler;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.Strings;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.sample.SampleResource;
+import org.opensearch.sample.resource.actions.rest.get.GetResourceAction;
+import org.opensearch.sample.resource.actions.rest.get.GetResourceRequest;
+import org.opensearch.sample.resource.actions.rest.get.GetResourceResponse;
+import org.opensearch.sample.resource.client.ResourceSharingClientAccessor;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.builder.SearchSourceBuilder;
+import org.opensearch.security.spi.resources.client.ResourceSharingClient;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+import org.opensearch.transport.client.node.NodeClient;
+
+import static org.opensearch.sample.utils.Constants.RESOURCE_INDEX_NAME;
+import static org.opensearch.security.spi.resources.FeatureConfigConstants.OPENSEARCH_RESOURCE_SHARING_ENABLED;
+import static org.opensearch.security.spi.resources.FeatureConfigConstants.OPENSEARCH_RESOURCE_SHARING_ENABLED_DEFAULT;
+
+/**
+ * Transport action for getting a resource
+ */
+public class GetResourceTransportAction extends HandledTransportAction<GetResourceRequest, GetResourceResponse> {
+    private static final Logger log = LogManager.getLogger(GetResourceTransportAction.class);
+
+    private final TransportService transportService;
+    private final NodeClient nodeClient;
+    private final Settings settings;
+
+    @Inject
+    public GetResourceTransportAction(
+        Settings settings,
+        TransportService transportService,
+        ActionFilters actionFilters,
+        NodeClient nodeClient
+    ) {
+        super(GetResourceAction.NAME, transportService, actionFilters, GetResourceRequest::new);
+        this.transportService = transportService;
+        this.nodeClient = nodeClient;
+        this.settings = settings;
+    }
+
+    @Override
+    protected void doExecute(Task task, GetResourceRequest request, ActionListener<GetResourceResponse> listener) {
+        ResourceSharingClient client = ResourceSharingClientAccessor.getResourceSharingClient();
```

Comment:
Following up to the conversation here: https://github.com/opensearch-project/security/pull/5281#discussion_r2054496615

I just now completely realized that the "optionality" of the security plugin can mean that the classes provided by the security plugin are actually not present in the classpath. 

However, then I do not understand how the code above can work without the security plugin present. If the security plugin is not present, the class `org.opensearch.security.spi.resources.client.ResourceSharingClient` will not be in the class path. I would expect, then class loading of `GetResourceTransportAction` to fail with a `NoClassDefFoundError`. Do I understand something wrong?

Reply:
Only if the code is trying to access the client at runtime when the class is not present on the runtime classpath. Its up to the resource plugins to ensure to only access it if provided.

The class can be referenced here because its available at compile time.

Reply:
But in the code above, it is accessed whenever the get resource action is executed. Even if the security plugin is not present.

Reply:
@DarshitChanpura for the sample plugin tests can we have 2 tracks?

1. Security installed
2. Security not installed

Reply:
NoopResourceSharingClient is assigned using the client accessor if security is disabled. SamplePlugin demonstrate a combination usage of the feature flag and using client directly. However, since Noop returns 501 on share and revoke, we use feature flag instead of check whether client should be called.

Reply:
I've removed NoopClient.

Reply:
> @DarshitChanpura for the sample plugin tests can we have 2 tracks?
Security installed
Security not installed

They are already present. `SampleResourcePluginSecurityDisabledTests` cover the disabled scenario and rest cover enabled scenarios along with feature enabled/disabled.

Reply:
There's a difference between security disabled and security absent. Did we test security absent - and that's in a real cluster, not with int tests?

Reply:
Yes, I can show a demo over slack if you'd like.

Reply:
Thanks 👍